kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -160,12 +168,16 @@ public void commit() throws IOException {
     } finally {
       buffer.clear();
       multiPartUpload = null;
-      close();
+      internalClose();
     }
   }
 
   @Override
   public void close() throws IOException {
+    internalClose();
+  }
+
+  private void internalClose() throws IOException {
```

Comment:
this seems like a tautology. why do we need the `internalClose`? Is it a leftover?

Reply:
It prevents from infinite recursion. `S3ParquetOutputStream` overrides `close()`, and calls `super.commit()` (i.e. `S3OutputStream#commit()`), but `S3OutputStream#commit()` will also call `close()`, which will refer to `S3ParquetOutputStream#close()`. Hence, the infinite recursion happened.

Unset `commit` flag before calling `super.commit()` can solve this problem as well, but I'm not sure which implementation is better. What do you think?